src/person.cpp

```diff
@@ -1617,6 +1664,73 @@ AnimateResult Guest::VisitRideOnAnimate(RideInstance *ri, TileEdge exit_edge)
 	return OAR_CONTINUE;
 }
 
+/** Pixel positions of guests sitting on a bench. */
+static XYZPoint16 _bench_pix_pos[4 /* TileEdge */][2 /* Left = 0, Right = 1 */] = {
+	{XYZPoint16(  0, 160, 0), XYZPoint16(  0,  96, 0)},
+	{XYZPoint16(160, 255, 0), XYZPoint16( 96, 255, 0)},
+	{XYZPoint16(255,  96, 0), XYZPoint16(255, 160, 0)},
+	{XYZPoint16( 96,   0, 0), XYZPoint16(160,   0, 0)},
+};
+
+AnimateResult Guest::InteractWithPathObject(PathObjectInstance *obj)
+{
+	const TileEdge edge = this->GetCurrentEdge();
+	if (obj->type == &PathObjectType::LITTERBIN && this->has_wrapper && obj->GetFreeBinCapacity(edge) > 0) {
+		/* Throw litter in the bin, then keep walking */
+		obj->AddItemToBin(edge);
+		this->has_wrapper = false;
+	} else if (obj->type == &PathObjectType::BENCH && this->nausea > 40 &&
+			(obj->GetLeftGuest(edge) == PathObjectType::NO_GUEST_ON_BENCH ||
+			obj->GetRightGuest(edge) == PathObjectType::NO_GUEST_ON_BENCH)) {
+		/* Sit down and remain there for a while. */
+		if (obj->GetRightGuest(edge) == PathObjectType::NO_GUEST_ON_BENCH) {
+			obj->SetRightGuest(edge, this->id);
+			this->pix_pos = _bench_pix_pos[edge][1];
+		} else {
+			obj->SetLeftGuest(edge, this->id);
+			this->pix_pos = _bench_pix_pos[edge][0];
+		}
+		this->activity = GA_RESTING;
+		this->StartAnimation(_guest_bench[edge]);
+		return OAR_OK;
+	} else if (this->happiness < 40) {
+		/* Smash something up, then keep walking. */
+		for (int8 dx = -2; dx <= 2; dx++) {
+			for (int8 dy = -2; dy <= 2; dy++) {
+				if (!IsVoxelstackInsideWorld(this->vox_pos.x + dx, this->vox_pos.y + dy)) continue;
+				for (int8 dz = -1; dz <= 1; dz++) {
+					const Voxel *vx = _world.GetVoxel(XYZPoint16(this->vox_pos.x + dx, this->vox_pos.y + dy, this->vox_pos.z + dz));
+					if (vx == nullptr) continue;
+					for (VoxelObject *o = vx->voxel_objects; o != nullptr; o = o->next_object) {
+						if (dynamic_cast<Guard*>(o) != nullptr) {
+							/* A security guard is nearby, so no vandalism just yet. */
+							return OAR_CONTINUE;
+						}
+					}
+				}
+			}
+		}
+
+		obj->Demolish(edge);
+	}
+	return OAR_CONTINUE;
+}
+
+AnimateResult Guest::ActionAnimationCallback()
+{
+	assert(this->activity == GA_RESTING);
+
+	if (this->food > 0 || this->drink > 0 || this->rnd.Uniform(255) > this->nausea) {
+		/* Remain sitting while eating, drinking, or nauseous. */
+		this->StartAnimation(_guest_bench[this->GetCurrentEdge()]);
+		return OAR_OK;
+	}
+
+	/* Get up and keep walking. */
+	this->activity == GA_WANDER;
```

Comment:
Nice github action

Reply:
You didn't see it?

![Screenshot from 2021-05-06 18-54-37](https://user-images.githubusercontent.com/3254232/117336530-969dfe80-ae9c-11eb-8901-eb4d7b5042f0.png)

GH action warns that an equality test doesn't do much :)

Reply:
Argh, I had overlooked that, and my compiler didn't bother to tell me